cpp/include/proxy_sequences.h

```diff
@@ -118,7 +118,30 @@ class UpdateUploadObjectFromFile final : public Update {
   const alba::Checksum *_cs_o;
 };
 
-/* TODO upload object from in memory blob */
+class UpdateUploadObject final : public Update {
+public:
+  UpdateUploadObject(const std::string &name,
+                     const std::shared_ptr<std::vector<char>> data,
+                     const alba::Checksum *cs_o)
+      : _name(name), _data(data), _cs_o(cs_o){};
+
+  void to(llio::message_builder &mb) const override {
+    mb.add_type(2);
+    llio::to(mb, _name);
+    uint32_t len = _data->size();
+    llio::to(mb, len);
+    mb.add_raw(_data->data(), len); // copies
+    if (_cs_o == nullptr) {
+      llio::to<boost::optional<const Checksum *>>(mb, boost::none);
```

Comment:
the problem is that `boost::none == boost::optional<X>(nullptr)` yields `false` so the serializer will try to deref the optional and SEGV in
```
template <typename X>
void to(message_builder &mb, const boost::optional<X> &xo) noexcept {
  if (boost::none == xo) {
    to(mb, false);
  } else {
    to(mb, true);
    to(mb, *xo);
  }
}
```